pages/index.js

```diff
@@ -1,33 +1,34 @@
-import { shoppingItems } from "../lib/shoppingItemsData.js";
-import { categories } from "../lib/categoriesData.js";
-
+import Link from "next/link";
 import styled from "styled-components";
 
-export default function ShoppingItemsList() {
-  const shoppingItemsWithCategoryColor = shoppingItems.map((shoppingItem) => {
-    const category = categories.find(
-      (category) => category.name === shoppingItem.category
-    );
-    const backgroundColor = category ? category.color : "white";
-    return {
-      ...shoppingItem,
-      backgroundColor,
-    };
-  });
-
+export default function ShoppingItemsList({ shoppingItemsWithCategoryColor }) {
+  console.log(
+    "shoppingItemsWithCategoryColor: ",
+    shoppingItemsWithCategoryColor
+  );
   return (
     <main>
-      <StyledH1> {shoppingItems.length} Shopping Items </StyledH1>
+      <StyledH1>
+        {" "}
+        {shoppingItemsWithCategoryColor.length} Shopping Items{" "}
+      </StyledH1>
       <StyledList>
-        {shoppingItemsWithCategoryColor.map((shoppingItem) => (
```

Comment:
I noticed that you are currently not using any components but rather hardcoding the pages. I think it might be nice to refactor the Items into their own component and then import it. So basically here you would return a component named Item or somehting like that instead of hardcoding it. This would make it easier to later reuse the component if you plan on having a second page with products. It would even be possible to reuse this in the [id].js and extend the styles. This is probably all optional at this point but thought I would still leave a comment. :)


```
{shoppingItemsWithCategoryColor.map((shoppingItem) => {
          return (
            <>
              <ListItem
                key={shoppingItem.id}
                $backgroundColor={shoppingItem.backgroundColor}
              >
                {shoppingItem.quantity} {shoppingItem.name}
                <CategoryBox>{shoppingItem.category}</CategoryBox>
                <StyledLink href={`/shoppingItems/${shoppingItem.id}`}>
                  Details
                </StyledLink>
              </ListItem>
            </>
          );
        })}
```

Reply:
Hey, thanks for your comment! I have decided not to use any components at this stage but you definitely have a point. I will most likely do that some time soon.